### PR TITLE
httphelper: Prevent multiple calls to response.WriteHeader when handling errors

### PR DESCRIPTION
**Pull request by jvatic**
refs #856

### CODE REVIEW

File: controller/jobs.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ func (c *controllerAPI) ListJobs(ctx context.Context, w http.ResponseWriter, req
 	app := c.getApp(ctx)
 	if strings.Contains(req.Header.Get("Accept"), "text/event-stream") {
 		if err := streamJobs(req, w, app, c.jobRepo); err != nil {
-			respondWithError(w, err)
+			httphelper.LogError(w, err)
```
Comment (titanous) [inline controller/jobs.go]: Can this return before we've written a 200 OK response?

File: pkg/httphelper/httphelper.go [hunk 2]
```diff
@@ -96,7 +96,16 @@ func (jsonError JSONError) Error() string {
 	return fmt.Sprintf("%s: %s", jsonError.Code, jsonError.Message)
 }
 
-func Error(w http.ResponseWriter, err error) {
+func logError(w http.ResponseWriter, err error) {
+	rw, ok := w.(*ResponseWriter)
+	if ok {
```
Comment (titanous) [inline pkg/httphelper/httphelper.go]:
```go
if rw, ok := w.(*ResponseWriter); ok {
```

**Comment by jvatic** [comment]
@titanous comments addressed

File: pkg/httphelper/httphelper.go [hunk 3]
```diff
@@ -109,23 +117,28 @@ func Error(w http.ResponseWriter, err error) {
...
 		}
 	}
+	return jsonError
+}
 
-	responseCode, ok := errorResponseCodes[jsonError.Code]
-	if !ok {
-		responseCode = 500
+func Error(w http.ResponseWriter, err error) {
+	jsonError := buildJSONError(err)
```
Comment (titanous) [inline pkg/httphelper/httphelper.go]: Looks like this can be moved inside the first branch of the if?

**Comment by jvatic** [comment]
@titanous comment addressed

**Comment by titanous** [comment]
LGTM

**Comment by jvatic** [comment]
(rebased)

**Pull request merged**